extract.py

```diff
@@ -0,0 +1,142 @@
+    :param change: single change between two trees
+    :return: does this blob contain non utf-8 characters
+    """
+    sha = (change.new.sha or change.old.sha)
+    try:
+        repo.get_object(sha).data.decode()
+    except UnicodeDecodeError as e:
+        return str(e).startswith("'utf-8' codec can't decode byte")
+    return False
+
+
+def into_lines(repo: Repo, sha: bytes) -> List[str]:
+    return repo.get_object(sha).data.decode().splitlines()
+
+
+def calc_diff(repo: Repo, change: TreeChange) -> Tuple[int, int]:
+    """
+    Get number of added and deleted lines
+
+    :param repo: repository with change
+    :param change: single change between two trees
+    :return: number of lines added/deleted
+    """
+    diff = difflib.Differ()
+    diffs = diff.compare(into_lines(repo, change.old.sha),
+                         into_lines(repo, change.new.sha))
+
+    added = 0
+    deleted = 0
+    for i in diffs:
+        if i[0] == "+" and i[1] != "+":
+            added += 1
+        elif i[0] == "-" and i[1] != "-":
+            deleted += 1
+
+    return added, deleted
+
+
+def process_only_blob(repo: Repo, sha: bytes, path: str, change_dict: defaultdict, missing: str, existing: str):
+    """
+    Write number of added and deleted lines for a file if it was created or deleted
+
+    :param repo: repository with change
+    :param sha: sha of blob
+    :param path: path to file
+    :param change_dict: dictionary of corresponding change
+    :param missing: nullified parameter for number of lines
+    :param existing: calculated number of lines parameter
+    """
+    change_dict[path][existing] = len(into_lines(repo, sha))
+    change_dict[path][missing] = 0
+    change_dict[path]["blob_id"] = str(repo.get_object(sha).id.decode()) if missing == "deleted" else str(None)
+
+
+def get_diff(repo: Repo, entry: WalkEntry) -> Dict[str, Dict]:
+    """
+    Get information about commit such as:
+        number of added lines for each file
+        number of deleted lines for each file
+        blob_id for each file
+
+    :param repo: pending repository
+    :param entry: commit object
+    :return: info in form of dict
+    """
+    res = defaultdict(dict)
+    for change in entry.changes():
+        if is_binary(repo, change):
+            continue
+
+        path = (change.new.path or change.old.path).decode()
+        try:
+            if change.old.sha is None:
+                process_only_blob(repo, change.new.sha, path, res, "deleted", "added")
+            elif change.new.sha is None:
+                process_only_blob(repo, change.old.sha, path, res, "added", "deleted")
+            else:
+                res[path]["added"], res[path]["deleted"] = calc_diff(repo, change)
+                res[path]["blob_id"] = str(repo.get_object(change.new.sha).id.decode())
+
+        # Handling corrupted files
+        except UnicodeDecodeError as e:
+            logger.error(f"exception in repository: {repo.path}, file: {path}, cause: {e}")
```

Comment:
Let's start error log with capital letter
```suggestion
            logger.error(f"Exception in repository: {repo.path}, file: {path}, cause: {e}")
````